tree/ntuple/v7/inc/ROOT/RNTupleWriteOptions.hxx

```diff
@@ -25,6 +25,12 @@
 namespace ROOT {
 namespace Experimental {
 
+class RNTupleWriteOptions;
+
+namespace Internal {
+void SetMaxKeySize(RNTupleWriteOptions &options, std::uint64_t maxKeySize);
```

Comment:
Do we actually need a class or are we ok with a specific namespace? (e.g. `Internal::RNTupleWriteOptionManipulation::SetMaxKeySize`)
In general I tend to prefer free functions whenever the functionality is "obvious" as they tend to be easier to use and understand (no hidden state)

Reply:
I guess the namespace would work, too. I'd still have a preference for a class (can be stateless and with static methods) because then there is for sure only one place where all the setters are defined. But in the end that's up to you.

Reply:
In the end I went with a class because it's also easier to friend-declare than separate functions